src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Helpers.cs

```diff
@@ -36,10 +36,17 @@ private static void ToHexArrayUpper(byte[] bytes, Span<char> chars)
 
         // Decode a hex string-encoded byte array passed to various X509 crypto api.
         // The parsing rules are overly forgiving but for compat reasons, they cannot be tightened.
-        public static byte[] DecodeHexString(this string s)
+        public static byte[] DecodeHexString(this string hexString)
         {
             int whitespaceCount = 0;
 
+            ReadOnlySpan<char> s = hexString;
+
+            if (s.Length != 0 && s[0] == '\u200E')
```

Comment:
Nit: it'd be helpful for someone later reading the code if there were a comment about what this value is and what it's skipped